examples/run_nlp_trainer.py

```diff
@@ -0,0 +1,365 @@
+            save_folder = args.save_folder
+        else: # finetune only
+            ckpt_folder = args.load_path
+            save_folder = os.path.join(args.load_path, '..')
+
+        all_ckpts_list =  os.listdir(ckpt_folder)
+            
+    else:
+        if args.training_scheme == 'all':
+            ckpt_folder = f'{args.run_name}/{args.save_folder}'
+            save_folder = args.save_folder
+        else: # finetune only
+            ckpt_folder = args.load_path # load path should be full/path/to/folder/ containing all ckpts 
+            save_folder = args.run_name 
+
+        all_ckpts_list = get_all_s3_checkpoints(args.bucket, ckpt_folder)
+
+    # finetune on every pretrained checkpoint  
+    for ckpt_filename in all_ckpts_list:
+        if args.save_locally:
+            ckpt_load_path = os.path.join(ckpt_folder, ckpt_filename)
+            save_locally = True
+            # skip symlinks to existing checkpoints
+            if (os.path.islink(ckpt_load_path)):
+                continue 
+        else:
+            ckpt_load_path = ckpt_filename
+            save_locally = False
+
+            # skip symlinks to existing checkpoints
+            # if ('symlink' in ckpt_filename):
+            #     continue
+            
+        # TODO: PUT SEEDS BACK IN
+        finetune_tasks = ['cola', 'mnli'] # 'sst-2', 'qqp', 'qnli']#, 'mnli']
+        save_ckpts = [False, True] # False, False, False]#, True]
+        setup_finetuning_jobs(finetune_tasks, save_ckpts, ckpt_load_path, ckpt_filename, save_folder, glue_metrics, wandb_logger=args.wandb_logger, save_locally=save_locally, bucket=args.bucket)
+
+        # finetune on inference tasks using last mnli checkpoint 
+        parent_ckpt = ckpt_filename # necessary for logging 
+        if args.save_locally:
+            # load checkpoints from where mnli saved them
+            ft_ckpt_folder = os.path.join(save_folder, 'mnli')  
+            ckpt_filename =  sorted(os.listdir(ft_ckpt_folder))[0]
+            ckpt_load_path = os.path.join(ft_ckpt_folder, ckpt_filename)
+            save_locally = True
+        else:
+            ckpt_filename = get_all_s3_checkpoints(args.bucket, f'{args.run_name}/mnli')[-1] # get latest checkpoint
+            ckpt_load_path = ckpt_filename
+            save_locally = False
+
+        mnli_finetune_tasks = ['rte', 'mrpc', 'stsb']
+        save_ckpts = [False, False, False]
+        # delete finetuning head to reinitialize number of classes 
+        setup_finetuning_jobs(mnli_finetune_tasks, save_ckpts, ckpt_load_path, ckpt_filename, args.save_folder, glue_metrics, 
+            wandb_logger=args.wandb_logger, save_locally=save_locally, bucket=args.bucket, parent_ckpt=parent_ckpt, load_ignore_keys=["state/model/model.classifier*"], )  # type: ignore
+
+def _main() -> None:
+    warnings.formatwarning = _warning_on_one_line
+
+    if len(sys.argv) == 1:
+        sys.argv = [sys.argv[0], '--help']
+    
+    args = get_args()
```

Comment:
Could we add some asserts or exception raising here on default values? For example, raising an exception if the user has specified `training_scheme = finetune` but no path to pretrained checkpoints, etc.